---
 lldb/source/API/CMakeLists.txt |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: llvm-toolchain-snapshot_19~++20240509085104+fd1bd53ba5a0/lldb/source/API/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_19~++20240509085104+fd1bd53ba5a0.orig/lldb/source/API/CMakeLists.txt
+++ llvm-toolchain-snapshot_19~++20240509085104+fd1bd53ba5a0/lldb/source/API/CMakeLists.txt
@@ -36,7 +36,12 @@ add_custom_command(
 add_custom_target(lldb-sbapi-dwarf-enums
   DEPENDS ${sb_languages_file})
 
-add_lldb_library(liblldb SHARED ${option_framework}
+set(output_name lldb)
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+  set(output_name liblldb)
+endif()
+
+add_lldb_library(liblldb SHARED OUTPUT_NAME ${output_name} SONAME
   SBAddress.cpp
   SBAttachInfo.cpp
   SBBlock.cpp
@@ -188,7 +193,7 @@ endif()
 
 set_target_properties(liblldb
   PROPERTIES
-  VERSION ${LLDB_VERSION}
+  VERSION 1
 )
 
 target_compile_definitions(liblldb PRIVATE LLDB_IN_LIBLLDB)
@@ -254,12 +259,12 @@ elseif (LLDB_EXPORT_ALL_SYMBOLS)
   set_target_properties(liblldb_exports PROPERTIES FOLDER "lldb misc")
 endif()
 
-if (NOT MSVC)
-  set_target_properties(liblldb
-    PROPERTIES
-    OUTPUT_NAME lldb
-  )
-endif()
+#if (NOT MSVC)
+#  set_target_properties(liblldb
+#    PROPERTIES
+#    OUTPUT_NAME lldb
+#    )
+#endif()
 
 # The Clang expression parser in LLDB requires the Clang resource directory to function.
 if (TARGET clang-resource-headers)
